scanpy/plotting/_tools/scatterplots.py

```diff
@@ -820,24 +820,37 @@ def spatial(
     If `show==False` a :class:`~matplotlib.axes.Axes` or a list of it.
     """
     if library_id is _empty:
-        library_id = next((i for i in adata.uns['spatial'].keys()))
+        try:  # check if it's Visium or other spatial data
+            library_id = next((i for i in adata.uns['spatial'].keys()))
+            spatial_data = adata.uns['spatial'][library_id]
+            if img_key is _empty:
+                img_key = next(
+                    (k for k in ['hires', 'lowres'] if k in spatial_data['images']),
+                    None,
+                )
+            elif img_key is None:  # invert coordinate if img_key is None
+                adata.obsm["spatial"][:, 1] = np.abs(
+                    np.subtract(
+                        adata.obsm["spatial"][:, 1],
+                        np.max(adata.obsm["spatial"][:, 1]),
+                    )
+                )
```

Comment:
Plotting functions shouldn't mutate data

Reply:
yes good point. I should just create a temporary adata object?

Reply:
I think the plot should just get it's y-axis inverted if you're expecting coordinates for an image.

Alternatively, a copy of just the spatial coordinates could be made.